pkg/virt-controller/watch/vm_test.go

```diff
@@ -1386,6 +1392,58 @@ var _ = Describe("VirtualMachine", func() {
 			testutils.ExpectEvent(recorder, SuccessfulDeleteVirtualMachineReason)
 		})
 
+		It("should add controller finalizer if VirtualMachine does not have it", func() {
+			vm, _ := DefaultVirtualMachine(false)
+			vm.Finalizers = nil
+
+			addVirtualMachine(vm)
+
+			shouldExpectVMFinalizerAddition(vm)
+			vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Return(vm, nil)
+
+			controller.Execute()
+		})
+
+		It("should add controller finalizer only once", func() {
+			//DefaultVirtualMachine already set finalizer
```

Comment:
nit: you can delete this comment and expect the finalizer to be an element of vm.Finalizers. This will both serve as documentation and provide a guarantee.

Reply:
Done